rules/aip0142/time_field_type.go

```diff
@@ -27,13 +27,23 @@ import (
 var fieldType = &lint.FieldRule{
 	Name: lint.NewRuleName(142, "time-field-type"),
 	LintField: func(f *desc.FieldDescriptor) []lint.Problem {
+		tokens := strings.Split(f.GetName(), "_")
+		suffix := tokens[len(tokens)-1]
+		typeName := utils.GetTypeName(f)
+		// Fast-path for civil dates and times
+		// https://aip.dev/142#civil-dates-and-times
+		switch {
+		case suffix == "date" && typeName == "google.type.Date":
+			return nil
+		case suffix == "time" && (typeName == "google.type.TimeOfDay" || typeName == "google.type.DateTime"):
+			return nil
+		}
 		suffixes := stringset.New(
 			"date", "datetime", "ms", "msec", "msecs", "millis", "nanos", "ns",
 			"nsec", "nsecs", "sec", "secs", "seconds", "time", "timestamp", "us",
 			"usec", "usecs",
 		)
-		tokens := strings.Split(f.GetName(), "_")
-		if suffixes.Contains(tokens[len(tokens)-1]) && utils.GetTypeName(f) != "google.protobuf.Timestamp" {
+		if suffixes.Contains(suffix) && typeName != "google.protobuf.Timestamp" {
```

Comment:
```suggestion
var allowed := stringset.New(
  "google.protobuf.Timestamp",
  "google.type.Date",
  "google.type.DateTime",
  "google.type.TimeOfDay",
)
if suffixes.Contains(suffix) && !allowed.Contains(typeName) {
```

...and then remove the switch statement.

Reply:
Good point! Fixed